server/migrations/20220810185000-scope-provider-id-uniqueness-to-team.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+module.exports = {
+  async up (queryInterface, Sequelize) {
+    await queryInterface.removeConstraint("authentication_providers", "authentication_providers_providerId_key");
+    await queryInterface.addConstraint("authentication_providers", {
+      type: 'unique',
+      fields: ["providerId", "teamId"],
+      name: "authentication_providers_providerId_teamId_uk"
+    });
+  },
+
+  async down (queryInterface, Sequelize) {
+    await queryInterface.removeConstraint("authentication_providers", "authentication_providers_providerId_teamId_uk");
+    await queryInterface.addConstraint("authentication_providers", {
+      type: 'unique',
+      fields: ["providerId"],
+      name: "authentication_providers_providerId_key"
+    });
+  }
+};
```

Comment:
🥦 this is a continuation of the logic change we started in previous PRs, where auth providers are scoped to the team instead of globally. All of the lookups are already team-scoped, so releasing the constraint doesn't change any existing behavior

Reply:
@tommoor this constraint prevents us from adding additional auth providers for a team. You can configure multiple auth providers such as Google, Slack, etc., but the login will fail if there is already an auth provider in the table for the team.

You receive the following error message after you authenticate with a provider
>Authentication failed – you do not have permission to access this workspace.

and the log message is not very enlightening either

`
{"error":"Validation error","level":"error","message":"Error during authentication","stack":"UnauthorizedError: Validation error\n    at InvalidAuthenticationError (/opt/outline/build/server/errors.js:44:34)\n    at accountProvisioner (/opt/outline/build/server/commands/accountProvisioner.js:59:54)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"}
`

Reply:
The constraint is for unique providerIds in a team, it does not prevent adding multiple unique auth providers – only duplicates.

Reply:
The `providerId` in the table is the configured domain e.g. `example.com` in our setup, Google + Gitlab as auth provider.

It can be verified in the auth provider implementations.
https://github.com/outline/outline/blob/main/plugins/google/server/auth/google.ts#L114
https://github.com/outline/outline/blob/main/plugins/oidc/server/auth/oidc.ts#L138

The constraint prevents the following combination from being added, which is absolutely valid and works as expected. Or is there something we are doing wrong?

```
id   | name   | providerId    | enabled | teamId   | createdAt          
--------------------------------------------------------------------------------
<id> | oidc   | example.com   | t       | <teamid> | 2022-09-15 11:25:56.539+00
<id> | google | example.com   | t       | <teamid> | 2023-07-06 14:14:09.233+00

```

Reply:
@tommoor have you seen my last comment?

Reply:
I would recommend just removing the constraint on your local instance in this case, thanks for raising.

Reply:
We have already done that, but thank you for taking a second look.